### ISSUE TITLE
api platform no longer installs seamlessly with Symfony

### ISSUE THREAD

**Issue by tacman** [symfony/symfony#58928]
### Symfony version(s) affected

7.1

### Description

**API Platform version(s) affected**: 4.0

A simple install with a Symfony webapp used to work but now throws an error.

### How to reproduce

```bash
symfony new MyApp  --webapp && cd MyApp
composer req api
bin/console make:entity -a Task -n
symfony server:start -d
symfony open:local --path=/api
```

![image](https://github.com/user-attachments/assets/82c27746-6774-4952-bdf6-d37bbdf751d0)

The files themselves, e.g. 
https://127.0.0.1:8029/assets/bundles/apiplatform/swagger-ui/swagger-ui-standalone-preset-4fec96b52efef25338c0ee2a2c2619e4.js

![image](https://github.com/user-attachments/assets/14f95b14-f030-4ff1-ac71-46fac2abc6c5)

I've reported this in api-platform as well: https://github.com/api-platform/api-platform/issues/2801

**Comment by laszlopokoradi** [comment]
I am facing the same issue, my "resolution" was to remove a **comment** from the \\vendor\\api-platform\\symfony\\Bundle\\Resources\\public\\graphiql\\graphiql.min.js file. 

This line contains a code example with `import('./path/to/domAnimations')`.

**Comment by tacman** [comment]
Yep, this code is commented out.

```js
// Asynchronous loading
/**
 * Used in conjunction with the `m` component to reduce bundle size.
 *
 * `m` is a version of the `motion` component that only loads functionality
 * critical for the initial render.
 *
 * `LazyMotion` can then be used to either synchronously or asynchronously
 * load animation and gesture support.
 *
 * ```jsx
 * // Synchronous loading
 * import { LazyMotion, m, domAnimations } from "framer-motion"
 *
 * function App() {
 *   return (
 *     <LazyMotion features={domAnimations}>
 *       <m.div animate={{ scale: 2 }} />
 *     </LazyMotion>
 *   )
 * }
 *
 * // Asynchronous loading
 * import { LazyMotion, m } from "framer-motion"
 *
 * function App() {
 *   return (
 *     <LazyMotion features={() => import('./path/to/domAnimations')}>
 *       <m.div animate={{ scale: 2 }} />
 *     </LazyMotion>
 *   )
 * }
 * ```
 *
 * @public
 */
```

Since assetmapper is not a javascript parser, it gets the paths it needs to resolve with a regex.  

Perhaps this commit?
https://github.com/symfony/asset-mapper/commit/21fe29b637ddaada739d3d6cd55694a7b95d2343

Although that probably wasn't it because it just added unicode support.

Until recently, this issue threw a warning in the javascript console, but now it won't even load.

**Comment by tacman** [comment]
The regex grabs it, then 

https://github.com/symfony/asset-mapper/commit/21fe29b637ddaada739d3d6cd55694a7b95d2343#diff-a6269ebc4116f5cfab61b63c4af5131c884e3f2267e442333760e1589de4b4eaR144-R169

I'll try to prepare a test maps some sort of fix.

**Comment by tacman** [comment]
I added a test, and am now attempting to fix the isCommentedOut method.

```php
    public static function provideCompileTests(): iterable
    {
        yield 'import_in_multiline_comment' => [
            'input' => <<<EOF
// Asynchronous loading
/**
 *
 * function App() {
 *   return (
 *     <LazyMotion features={() => import('./path/to/domAnimations')}>
 *       <m.div animate={{ scale: 2 }} />
 *     </LazyMotion>
 *   )
 * }
 * ```
 *
 * @public
 */
EOF
            ,
            'expectedJavaScriptImports' => [],
        ];

```

**Comment by tacman** [comment]
So it turned out to be a change in the asset-mapper _recipe_ to make import strict in development.

**Comment by adrianrudnik** [comment]
> So it turned out to be a change in the asset-mapper _recipe_ to make import strict in development.

The [recipe update](https://github.com/symfony/recipes/blob/ab0209c9b6fa6314403b39dae1a5b03cb722a46e/symfony/asset-mapper/6.4/config/packages/asset_mapper.yaml#L6) was the issue for me as well. Configuring this back to `warn` or `ignore` helped for now.

**Comment by zahitrios** [comment]
As @adrianrudnik said, the problem is the asset mapper, I removed this line from composer.json 
`"symfony/asset-mapper": "7.2.*"`

And then I did run `composer update`

**Comment by tacman** [comment]
No, that's not the right solution, because the application may need asset-mapper.  Instead, change 

```yaml
# config/packages/asset_mapper.yaml

framework:
    asset_mapper:
        # The paths to make available to the asset mapper.
        paths:
            - assets/
        missing_import_mode: warn
```
The problem is that the original recipe had the missing_import_mode set to strict.
